### PR TITLE
Better error handling for serdes failures during grpc calls

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
In a few different api calls here we were try catching the inner impl method, but not catching the exception and serializing it up if there was an error during the resulting serialize_value call itself. Add some additional try/catch wrappers for that.

### CODE REVIEW

File: python_modules/dagster/dagster/_grpc/server.py [hunk 1]
```diff
@@ -441,57 +462,78 @@ def ExternalNotebookData(self, request, _context) -> api_pb2.ExternalNotebookDat
...
 
-        instance_ref = args.instance_ref if args.instance_ref else self._instance_ref
-
-        serialized_data = serialize_value(
-            get_partition_config(
-                self._get_repo_for_origin(args.repository_origin),
-                args.partition_set_name,
-                args.partition_name,
-                instance_ref=instance_ref,
+        try:
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_grpc/server.py:493]: ^ [1]

**Review by alangenfeld — CHANGES_REQUESTED** [review]

File: python_modules/dagster/dagster/_grpc/server.py [hunk 2]
```diff
@@ -404,35 +408,52 @@ def ListRepositories(self, request, _context) -> api_pb2.ListRepositoriesReply:
...
-            else None,
-            repository_code_pointer_dict=loaded_repositories.code_pointers_by_repo_name,
-            entry_point=self._entry_point,
-            container_image=self._container_image,
-            container_context=self._container_context,
-            dagster_library_versions=DagsterLibraryRegistry.get(),
-        )
+        try:
+            loaded_repositories = check.not_none(self._loaded_repositories)
+            serialized_response = serialize_value(
```
[the remark this answers is not in the document]
Comment (alangenfeld) [reply python_modules/dagster/dagster/_grpc/server.py:413]: I think the consistent pattern is a good argument for sticking with the try / catch blocks

**Comment by alangenfeld** [comment]
thoughts on testing for this? its certainly possible to have tests for all the errors but not sure how much work that would take

**Comment by gibsondan** [comment]
yeah i have some sneaky tests in mind

File: python_modules/dagster/dagster/_grpc/server.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gibsondan) [reply python_modules/dagster/dagster/_grpc/server.py:413]: yeah i kinda just wanna maintain the invariant that if anything throws, even if surprising or unexpected, we get a stack trace over the wire

**Review by alangenfeld — APPROVED** [review]
nice

**Pull request merged**